### PR TITLE
feat: Added support for testing with Spring Security

### PR DESCRIPTION
**Pull request by mcollovati**
If Spring Security is present on classpath, then mock HttpServletRequest
is integrated with Spring SecurityContextHolder to provide Principal and
roles check support.
Also allows to use Spring `@WithMockUser`, `@WithAnonymousUser` and
`@WithUserDetails` annotations on tests.

Part of #1339

### CODE REVIEW

File: vaadin-testbench-unit/src/main/kotlin/com/vaadin/testbench/unit/internal/MockVaadin.kt [hunk 1]
```diff
@@ -437,3 +440,7 @@ private class MockPage(ui: UI, private val uiFactory: () -> UI, private val sess
         MockVaadin.createUI(uiFactory, session)
     }
 }
+
+fun interface MockRequestCustomizer {
+    fun apply(request: MockRequest)
+}
```
Comment (caalador) [inline vaadin-testbench-unit/src/main/kotlin/com/vaadin/testbench/unit/internal/MockVaadin.kt:446]: new line

File: vaadin-testbench-unit/src/main/java/com/vaadin/testbench/unit/mocks/MockWebApplicationContext.java [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+import org.springframework.context.MessageSourceResolvable;
+import org.springframework.context.NoSuchMessageException;
+import org.springframework.core.ResolvableType;
+import org.springframework.core.env.Environment;
+import org.springframework.core.io.Resource;
+import org.springframework.web.context.WebApplicationContext;
+
+/**
+ * An adapter to make a generic Spring ApplicationContext act like
+ * anWebApplicationContext.
```
Comment (caalador) [inline vaadin-testbench-unit/src/main/java/com/vaadin/testbench/unit/mocks/MockWebApplicationContext.java:34]:
```suggestion
 * a WebApplicationContext.
```

**Review by caalador — CHANGES_REQUESTED** [review]

File: vaadin-testbench-unit/src/main/kotlin/com/vaadin/testbench/unit/internal/MockVaadin.kt (hunk 1 above)
Comment (mcollovati) [reply vaadin-testbench-unit/src/main/kotlin/com/vaadin/testbench/unit/internal/MockVaadin.kt]: added

File: vaadin-testbench-unit/src/main/java/com/vaadin/testbench/unit/mocks/MockWebApplicationContext.java (hunk 2 above)
Comment (mcollovati) [reply vaadin-testbench-unit/src/main/java/com/vaadin/testbench/unit/mocks/MockWebApplicationContext.java]: fixed

**Review by caalador — APPROVED** [review]

**Pull request merged**